mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -94,10 +106,59 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       }
     }
 
+    // Obtain convolution parameters: padding / dialtion / stride.
+    auto leftPadH =
+        paddingAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto leftPadW =
+        paddingAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationH =
+        dilationsAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationW =
+        dilationsAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto strideH =
+        stridesAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto strideW =
+        stridesAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
```

Comment:
Could you use a more generic way to process linear attributes into an array, for example, below:

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/31d92f4c64ae6fa6b7c5d543f68b69300b4513ce/mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h#L165-L175

Reply:
@jerryyin Refactoring variable names is better handled in a subsequent PR. Could you help create a ticket for that?

Reply:
Agreed, I think there's some general duplication between high level lowering ops and tuning. Reducing those duplication could help overall test coverage. Will create issue.